apis/goharbor.io/v1alpha2/chartmuseum_types.go

```diff
@@ -0,0 +1,356 @@
+	Items           []ChartMuseum `json:"items"`
+}
+
+// ChartMuseumSpec defines the desired state of ChartMuseum.
+type ChartMuseumSpec struct {
+	harbormetav1.ComponentSpec `json:",inline"`
+
+	// +kubebuilder:validation:Optional
+	Log ChartMuseumLogSpec `json:"log,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Auth ChartMuseumAuthSpec `json:"auth,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Server ChartMuseumServerSpec `json:"server,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Disable some features
+	Disable ChartMuseumDisableSpec `json:"disable,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Cache stores
+	Cache ChartMuseumCacheSpec `json:"cache,omitempty"`
+
+	// +kubebuilder:validation:Required
+	Chart ChartMuseumChartSpec `json:"chart"`
+}
+
+type ChartMuseumServerSpec struct {
+	// +kubebuilder:validation:Optional
+	TLS *harbormetav1.ComponentsTLSSpec `json:"tls,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Type="string"
+	// +kubebuilder:validation:Pattern="([0-9]+h)?([0-9]+m)?([0-9]+s)?([0-9]+ms)?([0-9]+us)?([0-9]+µs)?([0-9]+ns)?"
+	// Socket timeout
+	ReadTimeout *metav1.Duration `json:"readTimeout,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Type="string"
+	// +kubebuilder:validation:Pattern="([0-9]+h)?([0-9]+m)?([0-9]+s)?([0-9]+ms)?([0-9]+us)?([0-9]+µs)?([0-9]+ns)?"
+	// Socket timeout
+	WriteTimeout *metav1.Duration `json:"writeTimeout,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default=20971520
+	// Max size of post body (in bytes)
+	MaxUploadSize *int64 `json:"maxUploadSize,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Value to set in the Access-Control-Allow-Origin HTTP header
+	CORSAllowOrigin string `json:"corsAllowOrigin,omitempty"`
+}
+
+type ChartMuseumChartSpec struct {
+	// +kubebuilder:validation:Optional
+	// Form fields which will be queried
+	PostFormFieldName ChartMuseumPostFormFieldNameSpec `json:"postFormFieldName,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern="https?://.*"
+	// The absolute url for .tgzs in index.yaml
+	URL string `json:"url"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=true
+	// Allow chart versions to be re-uploaded without ?force querystring
+	AllowOvewrite *bool `json:"allowOverwrite,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=false
+	// Enforce the chart museum server only accepts the valid chart version as Helm does
+	SemanticVersioning2Only bool `json:"onlySemver2"`
+
+	// +kubebuilder:validation:Required
+	Storage ChartMuseumChartStorageSpec `json:"storage"`
+
+	// +kubebuilder:validation:Optional
+	Index ChartMuseumChartIndexSpec `json:"index,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Repo ChartMuseumChartRepoSpec `json:"repo,omitempty"`
+}
+
+type ChartMuseumChartRepoSpec struct {
```

Comment:
Similar case to depth setting. It must be `1`.

```
# Multiple tenants
# Must be set with 1 to support project namespace
DEPTH=1
```

Reply:
I suggest t add comments to warn about that.
Following discussion on <https://github.com/goharbor/harbor-operator/pull/59#discussion_r467071357>